File: DALNEW/Query/PostalcodeQuery.vb
Imports System.Linq.Expressions
Imports BO
Public Class PostalcodeQuery
    Public Shared Function GetCityContainsQuery(search As String) As Expression(Of Func(Of PostalCode, Boolean))
        Return Function(f) f.Gemeente.ToLower().Contains(search)
    End Function
    Public Shared Function GetPostalCodeContainsQuery(search As String) As Expression(Of Func(Of PostalCode, Boolean))
        Return Function(f) f.Postcode.Contains(search)
    End Function
    Public Shared Function GetCountryQuery(countryID As Integer) As Expression(Of Func(Of PostalCode, Boolean))
        Return Function(f) f.CountryID = countryID
    End Function
    Public Shared Function GetPostalCodeStartsWithQuery(search As String) As Expression(Of Func(Of PostalCode, Boolean))
        Return Function(f) f.Postcode.StartsWith(search)
    End Function
End Class

File: DALCore/Query/PostalCodeQuery.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Linq.Expressions;
using System.Text;
using System.Threading.Tasks;
using DALCore.Models;
using BOCore;   

namespace DALCore.Query
{
  public  class PostalCodeQuery
    {
        public static Expression<Func<PostalCode, bool>> GetCityContainsQuery(string search)
        {
            return f => f.Gemeente.ToLower().Contains(search);
        }
        public static Expression<Func<PostalCode, bool>> GetPostalCodeContainsQuery(string search)
        {
            return f => f.Postcode.Contains(search);
        }
        public static Expression<Func<PostalCode, bool>> GetCountryQuery(int countryID)
        {
            return f => f.CountryId == countryID;
        }
        public static Expression<Func<PostalCode, bool>> GetPostalCodeStartsWithQuery(string search)
        {
            return f => f.Postcode.StartsWith(search);
        }
    }
}
